From 9aa9230a305e964b2417daea2b51e43b2dee607c Mon Sep 17 00:00:00 2001
From: Richard Purdie <richard.purdie@linuxfoundation.org>
Date: Tue, 27 Jun 2017 21:00:58 +0100
Subject: [PATCH] gtk-doc: Handle floating gtk-doc dependency

Allow the tests to be explicitly disabled to avoid floating dependnecy
issues. This is not really an issue with RSS but is on previous releases.

RP 2017/6/27
Upstream-Status: Pending
---
 configure.ac | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/configure.ac b/configure.ac
index 0491090..4270c88 100644
--- a/configure.ac
+++ b/configure.ac
@@ -82,6 +82,11 @@ if test "x$GCC" = "xyes"; then
 	fi
 fi
 
+AC_ARG_ENABLE([tests],
+	AS_HELP_STRING([--enable-tests],
+	[enable tests (default=yes)]),,
+	[enable_tests="yes"])
+
 dnl if glib is available we can enable the tests
 PKG_CHECK_MODULES(TEST_DEPS, [glib-2.0 >= 2.6.0 gobject-2.0 >= 2.6.0],
 	[	glib_prefix="`$PKG_CONFIG --variable=prefix glib-2.0`"
@@ -92,6 +97,11 @@ PKG_CHECK_MODULES(TEST_DEPS, [glib-2.0 >= 2.6.0 gobject-2.0 >= 2.6.0],
 		build_tests="no"
 	]
 )
+if test "x$enable_tests" != "xyes"; then
+	gtk_doc_use_libtool="no"
+	build_tests="no"
+fi
+
 AM_CONDITIONAL(GTK_DOC_USE_LIBTOOL, test -n "$LIBTOOL" -a x$gtk_doc_use_libtool = xyes )
 dnl this enables the rule in test/Makefile.am
 AM_CONDITIONAL(BUILD_TESTS, test x$build_tests = xyes)
